### PR TITLE
ARROW-7676: [Packaging][Python] Ensure that the static libraries are not built in the wheel scripts

### PR DESCRIPTION
**Pull request by kszucs**
It was already set for the manylinux201x builds.

### CODE REVIEW

**Comment by kszucs** [comment]
@github-actions crossbow submit wheel-manylinux1-*

**Comment by kszucs** [comment]
@github-actions crossbow submit wheel-osx-*

File: dev/tasks/python-wheels/osx-build.sh [hunk 1]
```diff
@@ -63,6 +63,7 @@ function build_wheel {
     mkdir build
     pushd build
     cmake -DARROW_BUILD_SHARED=ON \
+          -DARROW_BUILD_SHARED=OFF \
```
Comment (pitrou) [inline dev/tasks/python-wheels/osx-build.sh:66]: STATIC, not SHARED

**Comment by kszucs** [comment]
@github-actions crossbow submit wheel-osx-*

File: dev/tasks/python-wheels/osx-build.sh (hunk 1 above)
Comment (kszucs) [reply dev/tasks/python-wheels/osx-build.sh]: right :)

**Review by wesm — APPROVED** [review]
+1

**Pull request closed without merging**